client/sites/components/sites-dashboard.tsx

```diff
@@ -165,7 +165,9 @@ const SitesDashboard = ( {
 		sitesFilterCallback,
 		'all',
 		[ 'is_a4a_dev_site', 'site_migration' ],
-		[ 'theme_slug' ]
+		[ 'theme_slug' ],
+		// Don't fetch sites on narrow screens since it's not visible.
+		! selectedSite || isWide
```

Comment:
There's a couple things below this that leverage allSites list but I don't know if they're used, @fushar @cleacos I see you in the blames for some refactors around useShowSiteCreationNotice is this thing still valid or is it dead code? Creating a new site in testing took me to /home.

Reply:
The PR that added the line is https://github.com/Automattic/wp-calypso/pull/76612.

But that's weird; I just ran through the hosting flow but it landed me to /home, so I'm not sure what's happening. In any case, I think the logic still works -- the notice would still work because the new site ID is still not in `allSites` (which is empty now).

Reply:
It seems that the code was deleted in https://github.com/Automattic/wp-calypso/pull/79578. I'm unsure if it's intentional, but I believe we can remove this notice (triggered by `?new-site`), as the site creation notice is no longer in use. CC: @zaguiini